Example/AsyncReactorExample/Features/Repository/Search/RepositorySearchReactor.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  TestReactor.swift
+//  Rudi
+//
+//  Created by Dominik Arnhof on 14.04.23.
+//
+
+import Foundation
+import AsyncReactor
+import UIKit
+import Logging
+
+private let logger = Logger(label: "RepositorySearchReactor")
+
+class RepositorySearchReactor: AsyncReactor {
+    enum Action {
+        case onHidePrivateToggle
+        case enterQuery(String)
+        case load
+    }
+    
+    struct State {
+        var hidePrivate = false
+        var query = ""
+        var repositories: [Repository] = []
+        var isLoading = false
+    }
+    
+    @Published
+    private(set) var state: State
+    
+    @MainActor
+    init(state: State = State()) {
+        self.state = state
+        
+        lifecycleTask {
+            for await _ in await NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification).values {
+                await self.handleNotification()
+            }
+            
+            logger.debug("lifecycleTask cancelled")
+        }
+    }
+    
+    func action(_ action: Action) async {
+        switch action {
+        case .onHidePrivateToggle:
+            state.hidePrivate.toggle()
+        case .enterQuery(let query):
+            state.query = query
+            
+            try? await Task.sleep(for: .seconds(5))
+            
+            guard !Task.isCancelled else { return }
+            
+            logger.debug("search: \(query)")
```

Comment:
You should trigger the search request here. The idea behind the Task.sleep above is to achieve the debounce / throttle of the search requests. Maybe use a shorter delay than 5 seconds though :D